fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1415,7 +1415,10 @@ private RecordQueryPlan planScan(@Nonnull CandidateScan candidateScan,
     private RecordQueryPlan valueScan(@Nonnull CandidateScan candidateScan,
                                       @Nullable ScanComparisons scanComparisons,
                                       boolean strictlySorted) {
-        return planScan(candidateScan, IndexScanComparisons.byValue(scanComparisons), strictlySorted);
+        IndexScanType scanType = this.configuration.valueIndexOverScanNeeded(candidateScan.index.getName())
```

Comment:
A full scan is still a value scan; `index` might be null.

Reply:
@MMcM yeah I also noticed this and saw test failures with NPE. I have fixed this now.